src/SYCL-Graph/dynamicParamUpdateUSM.cpp

```diff
@@ -0,0 +1,88 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ **************************************************************************/
+
+#include <sycl/sycl.hpp>
+
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+using namespace sycl;
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((sycl_ext::nd_range_kernel<1>))
+void ff_1(int *PtrX, int *PtrY, int *PtrZ) {
+  size_t GlobalID =
+      ext::oneapi::this_work_item::get_nd_item<1>().get_global_id();
+  PtrX[GlobalID] += PtrY[GlobalID] * PtrZ[GlobalID];
+}
+
+int main() {
+  constexpr size_t Size = 1024;
+
+  queue Queue{};
+  auto Context = Queue.get_context();
+  auto Device = Queue.get_device();
+
+  // USM allocations for kernel input/output
+  int *PtrX = malloc_shared<int>(Size, Queue);
```

Comment:
Since shared USM is optional, should probably check that device supports it first and exit early if not. Same for any other tests using shared/host USM.